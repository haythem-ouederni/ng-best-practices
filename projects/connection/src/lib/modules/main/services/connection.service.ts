import {Injectable} from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class ConnectionService {
  constructor() {}

  calcultate(): string {
    return 'testing man';
  }
}
