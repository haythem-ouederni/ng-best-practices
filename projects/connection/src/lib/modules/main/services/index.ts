export * from './token';
export * from './connection.service';
