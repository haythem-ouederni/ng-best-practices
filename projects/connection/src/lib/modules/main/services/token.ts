import {InjectionToken} from '@angular/core';

export const CONNECTION_PATH = new InjectionToken<string>('CONNECTION_PATH');
