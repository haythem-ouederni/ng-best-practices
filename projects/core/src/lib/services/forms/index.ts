export * from './common-validation-messages.service';
