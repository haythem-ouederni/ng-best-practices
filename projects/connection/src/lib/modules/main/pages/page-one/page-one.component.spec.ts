import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {CONNECTION_PATH} from '../../services/token';
import {PageOneComponent} from './';

describe('PageOneComponent', () => {
  let component: PageOneComponent;
  let fixture: ComponentFixture<PageOneComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [PageOneComponent],
      providers: [
        {
          provide: CONNECTION_PATH,
          useValue: '',
        },
      ],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PageOneComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
