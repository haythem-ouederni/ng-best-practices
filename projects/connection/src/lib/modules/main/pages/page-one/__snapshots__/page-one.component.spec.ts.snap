// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageOneComponent should match snapshot 1`] = `
<cnx-page-one
  connectionPath=""
  connectionService={[Function ConnectionService]}
  display={[Function String]}
  linkToPageTwo={[Function String]}
>
  <p>
     Page One testing man
  
  </p><a
    href="/page-two"
    ng-reflect-router-link="//page-two"
  >
    Go to page two
  </a>
</cnx-page-one>
`;
